File: 38.js
function minDistinctInWindow(nums, k) {
    let map = new Map();
    let left = 0, distinct = 0, ans = Infinity;

    for (let right = 0; right < nums.length; right++) {
        map.set(nums[right], (map.get(nums[right]) || 0) + 1);
        if (map.get(nums[right]) === 1) distinct++;

        if (right - left + 1 === k) {
            ans = Math.min(ans, distinct);
            map.set(nums[left], map.get(nums[left]) - 1);
            if (map.get(nums[left]) === 0) distinct--;
            left++;
        }
    }
    return ans;
}
// Driver code
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split("\n");
const nums = input[0].split(" ").map(Number);
const k = parseInt(input[1], 10);
console.log(minDistinctInWindow(nums, k));

File: 38.py
from collections import defaultdict

def minDistinctInWindow(nums, k):
    freq = defaultdict(int)
    left = 0
    distinct = 0
    ans = float("inf")

    for right in range(len(nums)):
        freq[nums[right]] += 1
        if freq[nums[right]] == 1:
            distinct += 1

        if right - left + 1 == k:
            ans = min(ans, distinct)
            freq[nums[left]] -= 1
            if freq[nums[left]] == 0:
                distinct -= 1
            left += 1

    return ans
# Example usage:
nums = [1, 2, 1, 3, 4, 2, 3]
k = 4
print(minDistinctInWindow(nums, k)) # Output: 3
